app/controllers/cypress/user_creator_controller.rb

```diff
@@ -12,11 +14,13 @@ def create
       is_admin = (role == "admin")
 
       user = User.create(name: "#{role} Cypress", email: "#{role}@mampf.cypress",
-                         password: "cypress123", consents: true, admin: is_admin,
-                         locale: I18n.default_locale)
+                         name_in_tutorials: "#{role} Real Cypress Name",
+                         password: CYPRESS_PASSWORD, consents: true,
+                         admin: is_admin, locale: I18n.default_locale)
       user.confirm
 
-      render json: user.to_json, status: :created
+      additional_data = { password: CYPRESS_PASSWORD }
+      render json: user.as_json.merge(additional_data), status: :created
     end
```

Comment:
This is a good idea to return the cypress password such that we only have to define it once. However, in the `commands.js` for Cypress, you never actually use this returned parameter as far as I can tell.

Reply:
I think it is used, namely in the following lines:

```js
Cypress.Commands.add("login", (user) => {
  return cy.request({
    method: "POST",
    url: "/users/sign_in",
    form: true,
    failOnStatusCode: true,
    body: {
      "user[email]": user.email,
      "user[password]": user.password,
    },
  }).then((response) => {
    expect(response.status).to.eq(200);
  });
});
```

Without returning the additional password  in `user_creator_controller.rb`,  `user.password` is `undefined` and the login fails (see our basecamp ping from August 24). That's why I introduced this.

Reply:
moved to #696